device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -66,6 +66,7 @@
     private final static int MAX_MESSAGE_PAYLOAD_SIZE = 256*1024; //max IoT Hub message size is 256 kb, so amqp websocket layer should buffer at least that much space
     private final Boolean useWebSockets;
     private final Map<Integer, com.microsoft.azure.sdk.iot.device.Message> inProgressMessages = new ConcurrentHashMap<>();
+    private final Map<Integer, SubscriptionType> inProgressSubscriptionMessages = new ConcurrentHashMap<>();
```

Comment:
This new map will track outstanding subscription messages. I couldn't re-use the inProgressMessages map because there is no transport agnostic message to add to that map since we are creating the amqp message directly. Plus, messages in that map are expected to have a matching message in the IotHubTransport layer, which isn't the case for this special subscription message